WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/MainViewModel.kt

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.speaker
+
+import android.Manifest
+import android.app.Application
+import android.content.pm.PackageManager
+import android.media.AudioDeviceInfo
+import android.media.AudioManager
+import androidx.constraintlayout.motion.widget.MotionLayout
+import androidx.core.content.ContextCompat
+import androidx.core.content.getSystemService
+import androidx.lifecycle.AndroidViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+
+/**
+ * An [AndroidViewModel] driving the logic of [MainActivity].
+ */
+class MainViewModel(application: Application) : AndroidViewModel(application) {
+
+    /**
+     * The backing [MutableStateFlow] for the [appState].
+     */
+    private val appStateMutableStateFlow = MutableStateFlow<AppState>(AppState.Ready(transitionInstantly = true))
```

Comment:
I think some samples did that because they used the same name (without underscore) as the name of the public member. Here they have different names, which I think is fine.

Reply:
Yeah, I've used the `_` prefix before too, here I went with more explicit names since one was a `Channel` and two were `MutableStateFlow`s. I don't have a super strong preference either way though.

Reply:
SGTM, you can leave as is.